### PR TITLE
made message fields available & typescript definitions

### PR DESCRIPTION
**Pull request by teunmooij**
Implements #63

- In publish also accept all other message props in 2nd param and add those to the message
- Extended handler `properties` input field to also inclide remaining message fields

In v3.3.10 typescript definitions were added to [Systemic](https://github.com/guidesmiths/systemic). This PR also adds definitions to the Systemic Azure bus component.

### CODE REVIEW

**Comment by cressie176** [comment]
@teunmooij - this library came along after I left guidesmiths. The key ones I still have an interest in are rascal, systemic (plus runners + early components such as systemic-express, systemic-pg, systemic-redis and systemic-mongodb), confabulous (plus loaders), marv (plus drivers), whoosh, groundhog-day, outstanding, adbock-plus-crx, block-sequence (plus drivers), mr.robot, request-token, prepper & stashback.

File: lib/operations/topics/publish.js [hunk 1]
```diff
@@ -10,17 +10,17 @@ const encodingStrategies = {
...
-		messageId,
 		label,
-		contentType,
 		applicationProperties: {
 			contentEncoding,
-			...applicationProperties,
 			attemptCount: 0,
+			...applicationProperties,
 		},
+		...messageFields,
```
Comment (neodmy) [inline lib/operations/topics/publish.js:22]: Apart from the `messageId`, are there any other important properties?

**Review by neodmy — APPROVED** [review]
Apart from our pending conversation for the typescript part, the rest looks good. Thank you

File: lib/operations/topics/publish.js (hunk 1 above)
Comment (hamsaaldrobi) [reply lib/operations/topics/publish.js:22]: We should add the `deliveryCount` as well, but maybe for next PR

Comment (hamsaaldrobi) [reply lib/operations/topics/publish.js:22]: I think `deliveryCount` will be important for us

Comment (neodmy) [reply lib/operations/topics/publish.js:22]: That's on `applicationProperties` if I'm not mistaken

Comment (teunmooij) [reply lib/operations/topics/publish.js:22]: Looking at the `ServiceBusMessage` interface, I see a lot of fields people might want to use:
- contentType,
- partitionKey,
- sessionId
- replyToSessionId
- timeToLive
WE don't need them now, but someone else might. I don't see any good reason not to make these other fields available.

**Pull request closed without merging**